Code Restructure: More Pattern-matching

wli0503 opened the pull request:
-- Change in several helpers to a more pattern-oriented approach

wli0503 committed:
Code Restructure: More Pattern-matching

-- Change in several helpers to a more pattern-oriented approach

craigwaterman commented on lib/url_tincture.ex:
```diff
@@ -139,9 +155,10 @@ defmodule UrlTincture do
   """
   def httpish?(url) do
-    String.starts_with?(url, "http://") || String.starts_with?(url, "https://")
+    url =~ ~r/http[s]{0,1}:\/\//i
```
Please remove the case-insensitive flag from here, thanks

wli0503 committed:
Minor: More test coverage and minor changes to regular expression

- Added testcases for can_parse_safely?/1
- Remove 'case insensitive' flag from httpish?/1
